controllers/transactions.js

```diff
@@ -0,0 +1,36 @@
+const Transaction = require('../models/Transaction');
+const Appointment = require("../models/Appointment");
+
+// @desc: Post the e-receipt
+// @route: POST /api/v1/transactions/:appointmentId
+// @access: Private
+exports.addTransaction = async (req, res, next) => {
+   try {
+    req.body.appointment = req.params.appointmentId;
+
+    const appointment = await Appointment.findbyId(req.params.appointmentId);
+
+    if(!appointment) {
+      return res.status(404).json({
+        success: false,
+        message: `No appointment with the id of ${req.params.appointmentId}`,
+      });
+    }
+
+    // Add user Id to req body
+    req.body.user = req.user.id;
+
+    //Add campground Id to req body
+    const campground = appointment.campground;
+    req.body.campground = campground;
+
+    const transaction = await Transaction.create(req.body);
```

Comment:
I'm a bit confused as just creating new document in Transaction from req.body is not enough.

from the database schema the planned way of creating and updating a transaction document might look like this
1. get slip_image from req.body
2. insert slip_image into transactionSlip
3. insert various information (including ObjectId of the newly inserted slip_image doucment) into transaction where applicable

Thx.